asyncpg/protocol/protocol.pyx

```diff
@@ -725,6 +725,21 @@ cdef class BaseProtocol(CoreProtocol):
             self.last_query = None
             self.return_extra = False
 
+    cdef _on_notice(self, parsed):
+        # Check it here to avoid unnecessary `_field_map` iteration
+        cb = self.connection._notice_cb_set
+        if cb:
+            fields = apg_exc_base.PostgresMessage._field_map
+            message = {}
+            for k, v in parsed.items():
+                try:
+                    message[fields[k]] = v
+                except KeyError:
+                    # Ignore new unknown fields
+                    pass
+
+            self.connection._notice(message)
```

Comment:
A subclass of `PostgresMessage` should be returned here instead of a plain dict.